ecscollector/collector.go

```diff
@@ -0,0 +1,270 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package ecscollector implements a Prometheus collector for Amazon ECS
+// metrics available at the ECS metadata server.
+package ecscollector
+
+import (
+	"context"
+	"log"
+
+	"github.com/prometheus-community/ecs_exporter/ecsmetadata"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	numProcsDesc = prometheus.NewDesc(
+		"ecs_num_procs",
+		"Number of processes.",
+		labelKeys, nil)
+
+	cpuTotalDesc = prometheus.NewDesc(
+		"ecs_cpu_total",
+		"Total CPU usage.",
+		labelKeys, nil)
+
+	cpuUserDesc = prometheus.NewDesc(
+		"ecs_cpu_user",
+		"Total CPU usage by user space.",
+		labelKeys, nil)
+
+	cpuKernelDesc = prometheus.NewDesc(
+		"ecs_cpu_kernel",
+		"Total CPU usage by kernel space.",
+		labelKeys, nil)
+
+	cpuSystemDesc = prometheus.NewDesc(
+		"ecs_cpu_system",
+		"Total system CPU usage.",
+		labelKeys, nil)
+
+	cpuNumOnlineDesc = prometheus.NewDesc(
+		"ecs_cpu_num_online",
+		"Number of online CPUs.",
+		labelKeys, nil)
+
+	memUsageDesc = prometheus.NewDesc(
+		"ecs_mem_total",
+		"Total memory usage.",
+		labelKeys, nil)
+
+	memMaxUsageDesc = prometheus.NewDesc(
+		"ecs_mem_usage",
+		"Maximum memory usage.",
+		labelKeys, nil)
+
+	memLimitDesc = prometheus.NewDesc(
+		"ecs_mem_limit",
+		"Memory limit.",
+		labelKeys, nil)
+
+	networkRxBytesDesc = prometheus.NewDesc(
+		"ecs_network_rx_bytes",
+		"Network recieved in bytes.",
+		labelKeysWithNetwork, nil)
+
+	networkRxPacketsDesc = prometheus.NewDesc(
+		"ecs_network_rx_packets",
+		"Network packets recieved.",
+		labelKeysWithNetwork, nil)
+
+	networkRxDroppedDesc = prometheus.NewDesc(
+		"ecs_network_rx_dropped",
+		"Network packets dropped in recieving.",
+		labelKeysWithNetwork, nil)
+
+	networkRxErrorsDesc = prometheus.NewDesc(
+		"ecs_network_rx_errors",
```

Comment:
is `rx/tx` known in ECS world? Otherwise I'd go with node_exporter `receive` / `transmit` wording.

Reply:
Sounds good to use node exporter naming.